### ISSUE TITLE
Slim Sidebar has unexpected tab order

### ISSUE THREAD

**Issue by jessemenn** [wagtail/wagtail#8035]
### Issue Summary

Keyboard navigation into submenus of the page explorer doesn't work in the new slim sidebar. Using the chevron link to see a page's children shifts focus back to the main/sidebar menu's  next item. 

![menu-order](https://user-images.githubusercontent.com/1834167/155423692-70f10184-b7fa-414c-b967-86db0c4cbb85.gif)

### Steps to Reproduce

1.  Assuming a sidebar with Pages, Images, Documents etc... `tab` to and use the `enter` key on Pages in the sidebar
2.  Use `tab` to navigate to the chevron of a page with children and press `enter`.
3.  Press `tab` 
4.  Press `shift` + `tab` to return to the last focusable item of the children.

I'd expect focus to shift to the header of the submenu. 

* I have confirmed that this issue can be reproduced as described on a fresh Wagtail project: yes

### Technical details

* Python version: 3.9.x
* Wagtail version: 2.16.1, 
* Browser version: Chrome 98.0.4758.80 but presumably this will happen in all browsers.

**Comment by thibaudcolas** [comment]
Thank you for the report @jessemenn! Should hopefully be fixed in Wagtail 3.0 final via #8447.
